src/features/patient_view/clinical_information/ClinicalInformationTable.js

```diff
@@ -8,32 +9,87 @@ export class ClinicalInformationTable extends React.Component {
     }
 
     render() {
-        const arr = [];
-
-        // NOTE: we can use either naked bootstrap or a fancy bootstrap datatable control
-        // and get the same style.  That's the case for boostrap datatable control
-        this.props.data.forEach((item, i) => {
-            arr.push(<tr key={i}>
-                    <td>{ item.get(0) }</td>
-                    <td>{ item.get(1) }</td>
-                </tr>
+        if (this.props.data) {
+            const arr = [];
+
+            // NOTE: we can use either naked bootstrap or a fancy bootstrap datatable control
+            // and get the same style.  That's the case for boostrap datatable control
+            this.props.data.forEach((row, i) => {
+                arr.push(<tr key={i}>
+                         {row.map((item, j) => {
+                             return <td key={j}>{ item.toString() }</td>
+                         })}
+                    </tr>
+                );
+            });
+
+            return (
+                    <Table striped>
+                        <thead>
+                            <tr>
+                            {this.props.header.map((item ,i) => {
+                                return <th key={i}>{ item } </th>
+                            })}
+                            </tr>
+                        </thead>
+                          <tbody>{ arr }</tbody>
+                    </Table>
             );
-        });
-
-        return (
-                <Table striped>
-                    <thead>
-                        <tr><th>{ this.props.title1 }</th><th>{ this.props.title2 }</th></tr>
-                    </thead>
-                    <tbody>{ arr }</tbody>
-                </Table>
+        } else {
+            return "No data to display";
+        }
+    }
+}
+
+export function convertPatientDataToTable(patient) {
+    let rv;
+
+    rv = patient.get('clinicalData').map(x => Immutable.List([x.get('name'), x.get('value')]));
```

Comment:
Yes.  This is true also of the convertSampleDataToTable function below.  The code was very hard to read for two reasons, I think.  1. The format of the output was not human friendly (keyed objects) so it required that the reader understand the schema based on external knowledge.  2. maybe too much chaining and lookups.  I think in this case it would be better to forEach and build the desired output in a global variable for clarity.  Lets discuss.